go/ql/src/InconsistentCode/DeferredErrorCall.go

```diff
@@ -0,0 +1,15 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+func example() {
+	defer io.WriteString(os.Stdout, "All done!")
+
+	if _, err := io.WriteString(os.Stdout, "Hello"); err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Would it be possible to present a code example that preserves the behaviour we get from the usage of `defer`, while still handling any errors that that callee may produce i.e. what if I wanted to `defer` and also handle errors?

Reply:
Great question! It seems like there are techniques for doing this by deferring an anonymous function which captures and writes to an error variable, that is then returned by the parent function. I will modify the example to make use of that technique.